executor/src/main/java/io/kestra/executor/ExecutorService.java

```diff
@@ -1172,6 +1232,45 @@ public void addWorkerTaskResult(Executor executor, Supplier<FlowWithSource> flow
                     metricRegistry.tags(workerTaskResult)
                 )
                 .record(taskRun.getState().getDurationOrComputeIt());
+
+            if (!taskRun.getState().isFailed() && taskRun.getAssets() != null) {
+                AssetUser assetUser = new AssetUser(
+                    taskRun.getTenantId(),
+                    taskRun.getNamespace(),
+                    taskRun.getFlowId(),
+                    newExecution.getFlowRevision(),
+                    taskRun.getExecutionId(),
+                    taskRun.getTaskId(),
+                    taskRun.getId()
+                );
+
+                if (!taskRun.getAssets().getInputs().isEmpty() || !taskRun.getAssets().getOutputs().isEmpty()) {
```

Comment:
you can merge this if inside the enclosing if so if no inputs and no outputs the code is not run at all